main.tf

```
@@ -1,23 +1,144 @@
-terraform {
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = "~> 4.16"
-    }
+# Create a VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = "Main-VPC"
+  }
+}
+
+resource "aws_subnet" "public0" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "Public0-Subnet"
+  }
+}
+
+resource "aws_subnet" "public1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.2.0/24"
+
+  tags = {
+    Name = "Public1-Subnet"
+  }
+}
+
+resource "aws_subnet" "private0" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.3.0/24"
+
+  tags = {
+    Name = "Private0-Subnet"
   }
+}
+
+resource "aws_subnet" "private1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.4.0/24"
 
-  required_version = ">= 1.2.0"
+  tags = {
+    Name = "Private1-Subnet"
+  }
 }
+resource "aws_internet_gateway" "main" {
```

Newline is missing here
```suggestion
}

resource "aws_internet_gateway" "main" {
```